File: gateOne/daySeven/bankeBank/Python/Account.py
class Account:
    def __init__(self, first_name, last_name, pin, account_number):
        self._first_name = first_name
        self._last_name = last_name
        self._account_number = account_number
        self._pin = pin
        self._balance = 0.0

    def get_first_name(self):
        return self._first_name

    def get_last_name(self):
        return self._last_name

    def get_account_number(self):
        return self._account_number

    def get_balance(self):
        return self._balance

    def verify_pin(self, input_pin):
        return self._pin == input_pin

    def deposit(self, amount):
        if amount <= 0:
            raise ValueError("Deposit amount must be greater than zero")
        self._balance += amount

    def withdraw(self, amount, input_pin):
        if not self.verify_pin(input_pin):
            raise ValueError("Incorrect PIN")
        if amount <= 0:
            raise ValueError("Withdrawal amount must be greater than zero")
        if amount > self._balance:
            raise ValueError("Insufficient funds")
        self._balance -= amount

    def change_pin(self, old_pin, new_pin):
        if not self.verify_pin(old_pin):
            raise ValueError("Old PIN is incorrect")
        if old_pin == new_pin:
            raise ValueError("New PIN cannot be the same as the old PIN")
        if len(str(new_pin)) != 4:
            raise ValueError("New PIN must be exactly 4 digits")
        self._pin = new_pin

File: gateOne/daySeven/bankeBank/Java/Account.java
package daySeven.bankeBank.Java;

public class Account {
    private final String firstName;
    private final String lastName;
    private final String accountNumber;
    private int pin;
    private double balance;

    public Account(String firstName, String lastName, int pin, String accountNumber) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.accountNumber = accountNumber;
        this.pin = pin;
        this.balance = 0.0;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public String getAccountNumber() {
        return accountNumber;
    }

    public double getBalance() {
        return balance;
    }

    public boolean verifyPin(int inputPin) {
        return this.pin == inputPin;
    }

    public void deposit(double amount) {
        if (amount <= 0) {
            throw new IllegalArgumentException("Deposit amount must be greater than zero");
        }
        balance += amount;
    }

    public void withdraw(double amount, int inputPin) {
        if (!verifyPin(inputPin)) {
            throw new IllegalArgumentException("Incorrect PIN");
        }
        if (amount <= 0) {
            throw new IllegalArgumentException("Withdrawal amount must be greater than zero");
        }
        if (amount > balance) {
            throw new IllegalArgumentException("Insufficient funds");
        }
        balance -= amount;
    }

    public void changePin(int oldPin, int newPin) {
        if (!verifyPin(oldPin)) {
            throw new IllegalArgumentException("Old PIN is incorrect");
        }
        if (newPin == oldPin) {
            throw new IllegalArgumentException("New PIN cannot be the same as the old PIN");
        }
        if (String.valueOf(newPin).length() != 4) {
            throw new IllegalArgumentException("New PIN must be exactly 4 digits");
        }
        this.pin = newPin;
    }
}